componentified-todo-list/prdaneshi/to-do-list/src/index.js

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+
+const App = () => {
+  let [items, setItem] = useState([
+    { task: "first element", status: false },
+    { task: "second element", status: true },
+    { task: "third element", status: false },
+  ]);
+
+  const addToBotton = (event) => {
+    const newTask = {
+      task: inputValue,
+      status: false,
+    };
+
+    setItem([...items, newTask]);
+  };
+
+  const addToTop = (event) => {
+    const newTask = {
+      task: inputValue,
+      status: false,
+    };
+
+    setItem([newTask, ...items]);
+  };
+
+  const checkBoxClicked = (e, index) => {
+    console.log(index);
+    items[index].status = e.target.checked;
+    if (e.target.checked) {
+      const checked = items.splice(index, 1);
+      items.push(checked[0]);
+    }
+    setItem([...items]);
+  };
+
+  const DeleteTask = (e, index) => {
+    items.splice(index, 1);
+    setItem([...items]);
+  };
+
+  const [inputValue, setinputValue] = useState(0);
+  const InputValue = (e) => {
+    setinputValue(e.target.value);
+  };
+
+  const CreatList = () => {
+    const result = Object.entries(items).map((item) => (
+      <List
+        id={item[0]}
+        delete={(e) => DeleteTask(e, item[0])}
+        checkBoxClicked={(e, id) => checkBoxClicked(e, id)}
+        status={item[1].status}
+        task={item[1].task}
+      />
+    ));
+    return result;
+  };
+
+  return (
+    <div>
+      <input className="input" type="text" onChange={InputValue} />
+      <button onClick={addToBotton}> add to bottom</button>
+      <button onClick={addToTop}> add to top</button>
+      <ul>{CreatList()} </ul>
+    </div>
+  );
+};
+
+const List = (item) => {
+  return (
+    <li key={item.id}>
+      <i
+        className="fa fa-trash"
+        onClick={(e) => item.delete(e, item.id)}
```

Comment:
__[LOW]__  
You are messing up things here. `delete` prop doesn't accept a second arg, so this will suffice:
```jsx
onClick={(e) => item.delete(e)}
```
or event better:
```jsx
onClick={item.delete}
```